lib/stdlib/src/gen_server.erl

```diff
@@ -2353,15 +2334,16 @@ decode_msg(#server_data{parent = Parent} = ServerData, State, Msg, TRef, Debug,
             sys:handle_system_msg(Req, From, Parent, ?MODULE, Debug,
                                   [ServerData, State, TRef, Hib], Hib);
         {'EXIT', Parent, Reason} ->
+            cancel_timer(TRef),
```

Comment:
This was in before, but I removed it later because AFAICT `gen_statem` does not do that.

Reply:
Oh...  Here I saw an easy cleanup that could be done.  In `gen_statem` it is not that easy since there is no limit to the number of timers that may be running.

In both cases it is pointless to cancel the timer since the process is about to exit and will not receive any more, so there is no benefit in the call.

I will follow your example and remove (not add) it.  Thank you for pointing this out!